storage/storage.go

```diff
@@ -50,9 +54,23 @@ func (st *Storage) Set(key, value []byte) error {
 
 // Get gets the value for the given key.
 func (st *Storage) Get(key []byte) ([]byte, error) {
+	st.lock.Lock()
```

Comment:
This should be RLock() or instead of using a sync.RWMutex, use a sync.Mutex which is cheaper